testing/base/gtest_main.cpp

```diff
@@ -2,11 +2,34 @@
 // Exceptions. See /LICENSE for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+#include "testing/base/gtest_main.h"
+
 #include <gtest/gtest.h>
 
+#include <string>
+
+#include "common/check.h"
+#include "common/exe_path.h"
 #include "common/init_llvm.h"
 
+static bool after_main = false;
+static llvm::StringRef exe_path;
```

Comment:
Minor question, but had you considered:

```suggestion
static std::string* exe_path = nullptr;
```
 
i.e., rather than having a separate bool for tracking whether exe_path is initialized, use the pointer initialization.

Reply:
I didn't think of it, but I think I prefer the bool. I wanted to leave open the chance that we end up switching back to `llvm::SmallString` for the storage. And more generally, just avoid coupling the type of the global to the type of the storage in `main`.